src/features/wallet/root/ActivityCard/ActivityCard.tsx

```diff
@@ -59,17 +58,21 @@ const ActivityCard = forwardRef((props: Props, ref: Ref<BottomSheet>) => {
     },
   }))
 
-  const getData = useMemo((): AllTxns => {
-    if (isResetting) return []
-
+  const getData = useMemo(() => {
+    let data: (AnyTransaction | PendingTransaction)[] = txns
     if (filter === 'pending') {
-      return pendingTxns
+      data = pendingTxns
     }
+
     if (filter === 'all') {
-      return [...pendingTxns, ...txns]
+      data = [...pendingTxns, ...txns]
+    }
+
+    if (showSkeleton) {
+      return skeletonData(data.length)
```

Comment:
this is working great right now but one thought I had, would it be cleaner to cut over to an entirely different component with some fixed number of items in it for the skeleton view? that way we're not interweaving skeleton logic throughout our activity fetching code?

Reply:
Yeah, I definitely hear what you're saying. That likely would be cleaner. I'll give it a try and see how it feels.